### ISSUE TITLE
Some tests in io.votable failing on numpy-dev build

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#6850]
@mhvk , are there some changes in Numpy dev that could somehow trigger these? Any advice would be appreciated. Thank you.

Example: https://travis-ci.org/astropy/astropy/jobs/301866406

This one is due to some unexpected change in float precision:
```
_________________________ test_float_default_precision _________________________

    def test_float_default_precision():

        config = {'pedantic': True}

    

        field = tree.Field(

            None, name='c', datatype='float', arraysize="4",

            config=config)

        c = converters.get_converter(field, config=config)

>       assert (c.output([1, 2, 3, 8.999999], [False, False, False, False]) ==

                '1 2 3 8.9999990000000007')

E       AssertionError: assert '1 2 3 8.999999' == '1 2 3 8.9999990000000007'

E         - 1 2 3 8.999999

E         + 1 2 3 8.9999990000000007

astropy/io/votable/tests/converter_test.py:237: AssertionError
```

These are marked as `xfail` on certain condition but that condition did not apply here; However, could these also be a side effect of the same problem that caused the failure above?

```
_______________________________ test_regression ________________________________

tmpdir = local('/tmp/pytest-of-travis/pytest-0/test_regression0')

    @pytest.mark.xfail(str('legacy_float_repr'))

    def test_regression(tmpdir):

>       _test_regression(tmpdir, False)

astropy/io/votable/tests/vo_test.py:186:
...
        # If the lines happen to be different, print a diff

        # This is convenient for debugging

        for line in difflib.unified_diff(truth, output):

            sys.stdout.write(

                line.

                encode('unicode_escape').

>               replace('\\n', '\n'))

E           TypeError: a bytes-like object is required, not 'str'

astropy/io/votable/tests/vo_test.py:166: TypeError
```

```
_____________________ test_regression_python_based_parser ______________________

tmpdir = local('/tmp/pytest-of-travis/pytest-0/test_regression_python_based_p0')

    @pytest.mark.xfail(str('legacy_float_repr'))

    def test_regression_python_based_parser(tmpdir):

>       _test_regression(tmpdir, True)

astropy/io/votable/tests/vo_test.py:191: 
...
astropy/io/votable/tests/vo_test.py:166: TypeError
```

```
___________________________ test_regression_binary2 ____________________________

tmpdir = local('/tmp/pytest-of-travis/pytest-0/test_regression_binary20')

    @pytest.mark.xfail(str('legacy_float_repr'))

    def test_regression_binary2(tmpdir):

>       _test_regression(tmpdir, False, 2)

astropy/io/votable/tests/vo_test.py:196: 
...
astropy/io/votable/tests/vo_test.py:166: TypeError
```

**Comment by mhvk** [comment]
The first one is definitely numpy-dev related; see #6824. The bytes vs string thing seems a genuine mistake in the text (and I do think numpy became stricter here). The other two I'd have to look at in more detail; with numpy one can now pass a 'legacy' option to `np.set_printoptions` which may be needed...

p.s. numpy 1.14 is due fairly soon; I'll try to get to this later in the week.

**Comment by pllim** [comment]
Added a milestone so we don't forget about this. Please re-milestone as needed, or close if this is no longer an issue.

**Comment by pllim** [comment]
Potential fix in  #6959

**Comment by mhvk** [comment]
All fixed in #6959, so closing!

**Comment by pllim** [comment]
Thank you!
